resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -298,7 +298,7 @@ private[yarn] class YarnAllocator(
 
   // if a ResourceProfile hasn't been seen yet, create the corresponding YARN Resource for it
   private def createYarnResourceForResourceProfile(rp: ResourceProfile): Unit = synchronized {
-    if (!rpIdToYarnResource.contains(rp.id)) {
+    if (!rpIdToYarnResource.containsKey(rp.id)) {
```

Comment:
Which case covers this? Is it not covered before?

cc @tgravescs

Reply:
And Is this the only case in Spark?

Reply:
Looks like a clean fix, or at least it now does what this appears to want to do

Reply:
Sorry for late reply. I don't think there is a UT for this repeating case, it looks like map rpIdToYarnResource only serves as a cache for (increment_id ->  YarnResource), and DefaultProfile 0 is used in most cases, so the consequence should be limited.

I did test this patch on internal cluster, the repeated logs
> [main] INFO YarnAllocator: Resource profile 0 doesn't exist, adding it

are gone

Reply:
+CC @otterc